### DISCUSSION TITLE
不做“用户登录”背后的逻辑

### DISCUSSION THREAD

**Discussion by cshuo** [ChatGPTNextWeb/NextChat#702] [Q&A]
想了解一下为啥不做用户登录，是基于什么样的思考？

**Comment by realzsy** [comment]
做了以后，方便各种黄牛收割小白？

**Reply by cshuo** [reply]
> 做了以后，方便各种黄牛收割小白？

如果真是这个原因，未免有些因噎废食...毕竟只是个简单登录验证, 又不是加账号管理. Anyway, 只是个建议, main 分支没计划的话，就 fork 搞搞好了.

**Reply by Yidadaa** [reply]
不算因废噎食，你可以说一下加它有什么好处？

**Reply by cshuo** [reply]
> 不算因废噎食，你可以说一下加它有什么好处？

好处：对于私部 VPS 的用户来说, 网页服务暴露在互联中，会存在用户 (比如我) 不希望任何人都能白嫖 VPS 流量，那就可以通过验证登录把服务访问限制起来.  (与现有能力也不冲突, 比如登录后也可以选择共享 key 或指定 key 方式分享给同学朋友)

**Reply by Yidadaa** [reply]
你可以用很多种方法来实现访问限制，比如自己加个 NextAuth，或者使用 cloud flare 的 zero trust。

**Reply by greatlsd** [reply]
我希望搞个ip白名单，可以让指定的ip才能访问

**Comment by wangyendt** [comment]
This is an auto-sent e-mail. 
 
I am sorry that currently I could not reply your e-mail. I will reply you as soon as possible.
